napari/layers/tracks/_track_utils.py

```diff
@@ -0,0 +1,360 @@
+
+    @properties.setter
+    def properties(self, properties: Dict[str, np.ndarray]):
+        """ set track properties """
+        if not isinstance(properties, dict):
+            properties, _ = dataframe_to_properties(properties)
+
+        if 'track_id' not in properties:
+            properties['track_id'] = self.track_ids
+
+        # check the formatting of incoming properties data
+        self._properties = self._validate_track_properties(properties)
+
+    @property
+    def graph(self) -> Dict[str, list]:
+        """ dict {str: list}, graph representation of tracks """
+        return self._graph
+
+    @graph.setter
+    def graph(self, graph: Dict[str, list]):
+        """ set the track graph """
+
+        self._graph = self._validate_track_graph(graph)
+        # self.build_graph()
+
+    @property
+    def track_ids(self):
+        """ return the track identifiers"""
+        # return self.properties['track_id']
+        return self.data[:, 0].astype(np.uint16)
+
+    @property
+    def unique_track_ids(self):
+        """ return the unique track identifiers"""
+        return np.unique(self.track_ids)
+
+    def __len__(self):
+        """ return the number of tracks """
+        return len(self.unique_track_ids) if self.data is not None else 0
+
+    def _vertex_indices_from_id(self, track_id: int):
+        """ return the vertices corresponding to a track id """
+        return tuple(np.where(self.track_ids == track_id)[0])
+
+    def _validate_track_data(self, data: np.ndarray) -> np.ndarray:
+        """ validate the coordinate data """
+
+        if data.ndim != 2:
+            raise ValueError('track vertices should be a NxD array')
+
+        if data.shape[1] < 4 or data.shape[1] > 5:
+            raise ValueError('track vertices should be 4 or 5-dimensional')
+
+        if not all([idx.is_integer() for idx in data[:, 0].tolist()]):
+            raise ValueError('track id must be an integer')
+
+        if not all([t >= 0 for t in data[:, 1]]):
+            raise ValueError('track timestamps must be greater than zero')
+
+        return data
+
+    def _validate_track_properties(
+        self, properties: Dict[str, np.ndarray]
+    ) -> Dict[str, np.ndarray]:
+        """ validate the track properties """
+
+        for k, v in properties.items():
+            if len(v) != len(self.data):
+                raise ValueError(
+                    'the number of properties must equal the number of vertices'
+                )
+            # ensure the property values are a numpy array
+            if type(v) != np.ndarray:
+                properties[k] = np.asarray(v)
+
+        # if 'track_id' in properties:
+        #     track_idx = properties['track_id']
+        #     if not all(track_idx == np.maximum.accumulate(track_idx)):
+        #         raise ValueError('track_id should be monotonically increasing')
+        # else:
+        #     raise ValueError('track_id is missing from properties')
```

Comment:
This can be cut now we are going with the ID approach, or should we leave in? i.e. you could actually look in the properties dict see if `track_id` is present, if so use it and assume data is `T, (Z), Y, X` and it not assume data is `ID, T, (Z), Y, Z`.

This is certainly not required, but might be a nice way to get a "best of both worlds" without the API becoming more complex. On the other hand whenever we've tried to get more clever like this it's often created more difficulties because now other users/ tools can't assume exactly what data is they have to check the dict too, which becomes cumbersome.

Maybe best approach is to delete these lines and stick with `ID` as always axis 0